core/trino-main/src/main/java/io/trino/sql/planner/optimizations/HashGenerationOptimizer.java

```diff
@@ -106,7 +107,7 @@ public HashGenerationOptimizer(Metadata metadata)
     }
 
     @Override
-    public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator, WarningCollector warningCollector)
+    public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, SymbolAllocator symbolAllocator, PlanNodeIdAllocator idAllocator, WarningCollector warningCollector, TableStatsProvider tableStatsProvider)
```

Comment:
It would greatly help reviewing, if you had two commits

1. all the logic, which adds new `PlanOptimizer.optimize`  overload, delegating to the old one
   - this would change only those PlanOptimizers which use this new info, or which delegate to some other PlanOptimizer
2. remove the old method overload, which would add  `, TableStatsProvider tableStatsProvider` to many files (mechanical change)

Reply:
Makes sense. Separated them. Please feel free to squash them when merging if you feel appropriate.